gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/distributed/GfxdDistributionAdvisor.java

```diff
@@ -337,7 +337,8 @@ else if (p instanceof BridgeServerProfile) {
           else {
             isAdded = addMemberServer(m,
                 ThriftUtils.getHostAddress(bp.getHost(), bp.getPort())
-                    .setServerType(serverType), this.thriftServers, true);
+                    .setServerType(serverType).setIsCurrent(m.equals(GemFireStore.getMyId())),
```

Comment:
So the client which gets the hostaddress of the server at the connection time finds out whether it is connected to a server member or a locator member and based on that decides whether it has to do failover or not. Right?
So why can't on the server side we can just set this based on the vmkind information only?

Reply:
Server-side doesn't process this property. All handling of load-balance is done by client driver only which will decide whether to setup separate "control connection" to locator, and "data connections" to servers, use the getPreferredServer API, handle failure exceptions when appropriate to choose a new server using getPreferredServer on the "control connection" etc. Setting this property on the server-side will serve no purpose.

Reply:
Oh yes. Got confused. Thanks.